### DISCUSSION TITLE
ArgoCD helm error

### DISCUSSION THREAD

**Discussion by pdesai605** [argoproj/argo-cd#12111] [General]
I have setup ArgoCD in our Dev environment. 
We are trying to create application in ArgoCD but Getting this error due to secured secret zip file in our helm folder.
Helm structure:
**helm**
->files
->templates
    ---->deployment.yaml
    ---->sts.yaml
    ---->configmap.yaml
    ---->secret.yaml
    ---->service.yaml
    ---->other yaml files are also there
->values.yaml
->secret.yaml.7z (provided secret values in secured zip file)

**Error:**
Unable to create application: application spec for gatekeeper is invalid: InvalidSpecError: Unable to generate manifests in helm/application-db: rpc error: code = Unknown desc = `helm template . --name-template application --namespace preview-application-server-master --kube-version 1.23 --values .helm/application-db/values-preview.yaml --api-versions admissionregistration.k8s.io/v1 --api-versions admissionregistration.k8s.io/v1/MutatingWebhookConfiguration --api-versions admissionregistration.k8s.io/v1/ValidatingWebhookConfiguration --api-versions apiextensions.k8s.io/v1 --api-versions  v1/Namespace --api-versions v1/Node --api-versions v1/PersistentVolume --api-versions v1/PersistentVolumeClaim --api-versions v1/Pod --api-versions v1/PodTemplate --api-versions v1/ReplicationController --api-versions v1/ResourceQuota --api-versions v1/Secret --api-versions v1/Service --api-versions v1/ServiceAccount --include-crds` failed exit status 1: Error: template: postgresql-helm/templates/secrets.yaml:22:16: executing "postgresql-helm/templates/secrets.yaml" at <.Values.dbusers.patroni.password>: nil pointer evaluating interface {}.password Use --debug flag to render out invalid YAML

**Is there any way we can unzip the file from Argocd or we can read the zip file content from argocd?**

Can anyone help me one this?

**Comment by reginaelyse** [comment]
I'm not too familiar with this, but I wonder if you could possibly utilize a tool like kustomize and a custom script to unzip the file? I'm not sure that Argo CD provides built-in behavior for unzipping

**Reply by reginaelyse** [reply]
Hey @pdesai605, I brought up your issue in the weekly argocd contributor's call and we agreed that this isn't really possible and instead you should probably switch to using a secrets manager. The full discussion can be found [here](https://youtu.be/XYOQxRTqlEQ?t=354), please let us know if you have any further questions!

**Reply by pdesai605** [reply]
Thanks for your response.
